### PR TITLE
[c-ares] Update to 1.17.1.

### LINKED ISSUE #14828: [c-ares] update to 1.17.1
Library name: c-ares

New version number: 1.17.1

Other information that may be useful:

Release announcement, including change summary: https://c-ares.haxx.se/mail/c-ares-archive-2020-11/0008.shtml

### PR DESCRIPTION
**Pull request by geraldcombs**
Update c-ares to 1.17.1, the current version as of
2020-11-30.  The release announcement can be found at
https://c-ares.haxx.se/mail/c-ares-archive-2020-11/0008.shtml.
Fixes #14828.

### CODE REVIEW

File: ports/c-ares/portfile.cmake [hunk 1]
```diff
@@ -46,6 +46,7 @@ endif()
 vcpkg_copy_pdbs()
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
 
 # Handle copyright
 file(COPY ${SOURCE_PATH}/LICENSE.md DESTINATION ${CURRENT_PACKAGES_DIR}/share/c-ares)
```
Comment (NancyLi1013) [inline ports/c-ares/portfile.cmake:52]:
```suggestion
file(INSTALL ${SOURCE_PATH}/LICENSE.md DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
And remove this line:
`file(RENAME ${CURRENT_PACKAGES_DIR}/share/c-ares/LICENSE.md ${CURRENT_PACKAGES_DIR}/share/c-ares/copyright)`

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/c-ares/portfile.cmake [hunk 2]
```diff
@@ -5,8 +5,8 @@ endif()
 vcpkg_from_github(
```
[the remark this answers is not in the document]
Comment (geraldcombs) [reply ports/c-ares/portfile.cmake:3]: Done.

File: ports/c-ares/portfile.cmake (hunk 1 above)
Comment (geraldcombs) [reply ports/c-ares/portfile.cmake]: Done.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your PR @geraldcombs.

**Comment by strega-nil** [comment]
Thanks @geraldcombs !

**Pull request merged**